Isolation/test/src/IsolationAnalysis.cc

```diff
@@ -378,69 +458,93 @@ void IsolationAnalysis::fillLUTProgression(std::string option) {
+                         <<"  Estimated time left : "<< std::chrono::duration<double, std::milli>(t_end-t_start).count()*( maxEntriesForLUT - count)/(1e-9 + count)* 0.001
+                         <<std::endl;
+                }
+                // Loading the fit for correponding [ Eff , Et , Eta ]
+                TString fitName = "fit_pz_"+to_string(Int_Efficiency_Progression)+"_eta"+to_string(i)+"_e"+to_string(j);
+                TF1* currentFit = (TF1*) (WPFile->Get("Step1Histos/"+fitName));
 
+                if ( not currentFit)
+                {
+                    std::cout<<" Fit not found for "<< fitName<<" ! \n \t Exiting !!  \n";
+                    exit(1);
                 }
-                for (auto it : lutProgHistoMap_)
+                for(Int_t k = 0 ; k < lutnTTVec_.size()-1; k++)
                 {
-
-                    std::string tmp_string = it.first;
-                    std::vector<std::string> options;
-                    tokenize(tmp_string,options,":");
-                    Double_t minPt =  std::stod(options[1]);
-                    Double_t effLowMinPt = std::stod(options[2]);
-                    Double_t reach100pc= std::stod(options[3]);
-
-                    Double_t Efficiency_Progression = findEfficiencyProgression((lutIEtVec_[j]+lutIEtVec_[j+1])/2.0, minPt,effLowMinPt,reach100pc);
-                    if(Efficiency_Progression >= 0.9999) Efficiency_Progression = 1.0001;
-                    Int_t Int_Efficiency_Progression = int(Efficiency_Progression*100);
-                    TH3F* eff_histo;
-                    if(option == "do_2") {
-                        TString WPName = "Eff_";
-                        WPName +=std::to_string(Int_Efficiency_Progression);
-                        eff_histo = dynamic_cast<TH3F*>(WPFile->Get("Step1Histos/"+WPName));
-                        //std::cout<<WPName<<std::endl;
-                    }
-                    else
-                        eff_histo = IsoCut_PerBin[Int_Efficiency_Progression];
                     Int_t IsoCut_Progression = eff_histo->GetBinContent(i+1,j+1, k+1);
                     if(Int_Efficiency_Progression==100) IsoCut_Progression = 1000;
                     it.second->SetBinContent(i+1,j+1,k+1,IsoCut_Progression);
-                }
 
-                for(UInt_t iEff = 0 ; iEff < 101 ; ++iEff)
-                {
-                    TH3F* th;
-                    if(option == "do_2") {
-                        TString WPName1 = "Eff_";
-                        WPName1 +=TString::Itoa(iEff,10);
-                        th = dynamic_cast<TH3F*>(WPFile->Get("Step1Histos/"+WPName1));
-                    }
-                    else
-                        th= IsoCut_PerBin[iEff];
-                    Int_t IsoCut = th->GetBinContent(i+1,j+1, k+1);
+                    // Filling from the Fit extrapolation
+                    IsoCut_Progression = max(Int_t(currentFit->GetParameter(0) + currentFit->GetParameter(1) *k  ) , 1);
```

Comment:
resolved [comment](https://github.com/ats2008/L1EGCalibrations/pull/10#issuecomment-1473287492)
@sbaradia21 
please see if the max( ... ,  1) also makes sense

Reply:
From [here](https://github.com/ats2008/L1EGCalibrations/blob/main/Isolation/test/src/MakeEGIsoLUT.C#L159) it seems the LUT making process exports the IsoCut adding 1 to the actual values in the TH3 we have for each option.

so it should probably be max( ... , 0 )

Reply:
resolved in previous [commit](https://github.com/ats2008/L1EGCalibrations/pull/10/commits/9b23e27d83c53e256349c85b7252d3f19aecfc30)